src/figuresClasses.ts

```diff
@@ -1,19 +1,87 @@
-export interface Figure {
+function isLessThanZero(param: number): boolean {
+  return param <= 0;
+}
 
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
 }
 
-export class Triangle {
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
 
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if ([a, b, c].some(isLessThanZero)) {
+      throw new Error('Invalid input params!');
+    }
+
+    if ((a + b) <= c || (a + c) <= b || (b + c) <= a) {
+      throw new Error('It is not a triangle!');
+    }
+  }
 
+  getArea(): number {
+    const p: number = (this.a + this.b + this.c) / 2;
```

Comment:
What is "p"? if you use math definitions, where "p" is perimeter - in your variable only half of it. I mean: let's try to use more explicit naming)